codegen/src/node_ok.rs

```diff
@@ -4,7 +4,20 @@ use indoc::formatdoc;
 
 use crate::{message::MessageCodegen, Codegen, Indent};
 
-impl<'n> Codegen<'n> {
+pub trait NodeOk {
+    /// Name of the NodeOk function for the node.
+    fn node_ok_fn_name(&self, node: &str) -> String;
+    /// Declaration of the NodeOk function for the node.
+    fn node_ok_fn_decl(&self, node: &str) -> String;
+    /// Definition of the NodeOk function for the node.
+    fn node_ok_fn_def(&self, node: &str) -> String;
+    /// Declarations of NodeOk functions for the current node.
+    fn node_ok_fn_decls(&self) -> String;
+    /// Definitions of NodeOk functions for the current node.
+    fn node_ok_fn_defs(&self) -> String;
+}
+
+impl NodeOk for Codegen<'_> {
```

Comment:
We should just call each message_ok function here as needed rather than having the same logic in both. In each `node_ok_fn_def`, we can call the right `message_ok_fn_name`.